core/src/test/java/org/pentaho/platform/dataaccess/datasource/api/MetadataServiceTest.java

```diff
@@ -136,9 +139,9 @@ public void testGetMetadataDatasourceIds() throws Exception {
     mockSet.add( "domainId1" );
     mockMetadataIdsList.add( "domainId1" );
 
-    doReturn( true ).when( metadataService ).isMetadataDatasource( "domainId1" );
+    doReturn( 1 ).when( metadataService ).noOfThreads();
+    doReturn( "domainId1" ).when( metadataService ).isMetaDataSource( "domainId1" );
     doReturn( mockSet ).when( metadataService.metadataDomainRepository ).getDomainIds();
-
     List<String> response = metadataService.getMetadataDatasourceIds();
 
     verify( metadataService, times( 1 ) ).getMetadataDatasourceIds();
```

Comment:
there's no harm keeping this here, but we need to verify that `#getMetadataDatasourceIds()` gets called once here?